awx/main/notifications/webhook_backend.py

```diff
@@ -32,6 +37,7 @@ def send_messages(self, messages):
         sent_messages = 0
         if 'User-Agent' not in self.headers:
             self.headers['User-Agent'] = "Tower {}".format(get_awx_version())
+        self.headers['Authorization'] = base64.b64encode("{}:{}".format(self.username, self.password).encode())
```

Comment:
Python's `requests` will do the hard work for you 😄 

https://2.python-requests.org/en/master/user/authentication/#basic-authentication

```python
r = requests.post(
    "{}".format(m.recipients()[0]),
    auth=(self.username, self.password),
    ...
)